### PR TITLE
getting interrupt information from cu, not relying on cu_isr

### PR DESCRIPTION
**Pull request by Himanshu-xilinx**

### CODE REVIEW

**Comment by chvamshi-xilinx** [comment]
retest this please

**Comment by mamin506** [comment]
Read IP layout in scheduler is not ideal but acceptable.
Let's keep it here and we could clean it up at next release.

**Comment by xdavidz** [comment]
> Read IP layout in scheduler is not ideal but acceptable.
> Let's keep it here and we could clean it up at next release.

If we are using xclbin.h, I would prefer move all those logic into a separate file like zocl_xclbin.c in which we include the xclbin.h only. The scheduler should not include the xclbin.h and else where. We will get this fixed in next release.

File: src/runtime_src/core/edge/drm/zocl/sched_exec.c [hunk 1]
```diff
@@ -662,6 +662,26 @@ init_cus(struct sched_cmd *cmd)
...
+get_cus_support_intr(struct drm_zocl_dev *zdev)
+{
+	struct ip_data *ip;
+        int i;
+        if (!zdev->ip)
+                return false;
+
+        for (i = 0; i < zdev->ip->m_count; ++i) {
+                ip = &zdev->ip->m_ip_data[i];
+                if(!(ip->properties & 0x1)) {
```
Comment (mamin506) [inline src/runtime_src/core/edge/drm/zocl/sched_exec.c]: Confirmed with system compiler, if the CU has a interrupt pin, the bit 0 of properties will be set to 1. This is the correct way to determine if a CU supports interrupt.

**Comment by mamin506** [comment]
> If we are using xclbin.h, I would prefer move all those logic into a separate file like zocl_xclbin.c in which we include the xclbin.h only. The scheduler should not include the xclbin.h and else where. We will get this fixed in next release.

Yes. Let's do code refactoring for the next release. I think we should have scheduler, CU, xclbin, sysfs etc. components in zocl driver.

**Comment by chvamshi-xilinx** [comment]
Thanks @mamin506  and @xdavidz , We have verified multiple cases and enabled and merged this change, 
We will move it to separate file in master.

**Comment by david2011qq** [comment]
> Thanks @mamin506 and @xdavidz , We have verified multiple cases and enabled and merged this change,
> We will move it to separate file in master.

Thanks for the comment, the xclbin related code are not just from here. I have done the code change with context switch for versal. Will add you as a code reviewer. Thanks!

**Pull request merged**